### PR TITLE
Enable other distros for building

### PR DESCRIPTION
**Pull request by ellismg**
This builds on top of @joperezr recent pull requests, because he went through the trouble of updating buildtools and the CLI for me.  Once his is merged, this could go it.

I am still doing validation on the new platforms in docker, but Ubuntu 16.04 worked as expected.  Opening now to get a CI run.

/cc @gkhanna79 @weshaggard

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -209,25 +209,33 @@ isMSBuildOnNETCoreSupported()
     # This needs to be updated alongwith corresponding changes to netci.groovy.
     __isMSBuildOnNETCoreSupported=0
```
Comment (weshaggard) [inline build.sh]: Does it make more since to invert this default and assume it is supported and detect failure in some other way?

Comment (ellismg) [inline build.sh]: We could certainly change the condition to default to true and then set it to false in a `*)` of the case (have all the other cases be empty).

I think the right long term thing is to probably just try `init-tools.sh` and if that works assume we can build managed code.

**Comment by joperezr** [comment]
Just as FYI, mi PR is merged

**Comment by briansull** [comment]
Please update the coreclr intro page to announce the support for the other distros and for Ubunto 16

File: build.sh (hunk 1 above)
Comment (ellismg) [inline build.sh]: I filed #5318 to follow up here.

**Comment by weshaggard** [comment]
LGTM

**Comment by ellismg** [comment]
Needed to rebase again, once this is green I will merge and port the relevant stuff to release/1.0.0

**Comment by pgavlin** [comment]
This works fine on my Fedora machines, FWIW. Thanks @ellismg :)

**Pull request merged**